timemachine/fe/energy_decomposition.py

```diff
@@ -47,8 +48,12 @@ def get_batch_u_fns(
     batch_u_fns: List[Batch_u_fn] = []
     for p, pot in zip(params, pots):
 
-        def batch_u_fn(xs, boxes, pot_impl, pot_params):
-            coords = np.array([x for x in xs])
+        def batch_u_fn(xs: NDArray | StoredArrays, boxes, pot_impl, pot_params):
+            # If the coordinates are already an in-memory numpy array, don't create a copy
+            if isinstance(xs, StoredArrays):
+                coords = np.array([x for x in xs])
```

Comment:
nit: I think
```suggestion
                coords = np.array([x for x in xs])
```
```suggestion
                coords = np.array(xs)
```

is sufficient

Reply:
Fixed this case and one other in https://github.com/proteneer/timemachine/pull/1323/commits/c568f9b8859bb095038cfe709d196248e038c5c5